### ISSUE TITLE
suggest to add two new routes and link dose form to routes

### ISSUE THREAD

**Issue by tiozab** [OHDSI/Vocabulary-v5.0#860]
# Instructions

**0. Create the initial issue.**

We have worked on a route hierarchy that is tidy, hierarchical and linked to dose form. We think the community would greatly benefit from seeing this route hierarchy replacing the existing route domain. @cgreich 

**1. Read the documentation.**

Please familiarize yourself with the documentation [here](https://github.com/OHDSI/Vocabulary-v5.0/wiki/Community-contribution) prior to initiating the submission.
DONE!

**2. Choose the category of submission.**

Choose the category of your submission and download the corresponding template:

- Template 5: Modify attributes of existing concept(s): domain, name, etc.;
DOWNLOADED!

**3. Fill the template.**

Please dowload the corresponding template from [this folder](https://drive.google.com/drive/u/1/folders/1D7hSGKQkWgJCAKXDNOjfoFcnRca0-57w) and fill it. Instructions and examples are provided inside. If questions occurr use comments in the current issue to contact the Vocabulary Team. Make sure you filled the checklist!

**4. Upload the template.**

Please go to [this folder](https://drive.google.com/drive/u/1/folders/1rWPvto3zdS7F4QrhkpL3JXzDfm60tGOa), create a subfolder with your contribution (name it as GitHubIssueNumber_ContributionName_YourName) and upload your contribution there.
DONE!

**5. Reference the template in this issue.**

Will edit, need the GitHubIssueNumber first.
https://drive.google.com/drive/u/0/folders/1jEH-ZrxrywQvNMhgJnmWeX9xSIdPpBD5

**6. Get the feedback.**

'Modify content' contributions are  reviewed by the Vocabulary Team. We will look at the submission. We will let you know if further discussion with the community is needed.

**Comment by tiozab** [comment]
I would like to add a bit more information, we made an OHDSI poster about it and presented at OHDSI Europe (we will also present this at the OHDSI US symposium). Please see the pdf below. 

[poster_dose_to_route_TB.pdf](https://github.com/OHDSI/Vocabulary-v5.0/files/12170467/poster_dose_to_route_TB.pdf)

**Comment by aostropolets** [comment]
Hi Theresa! Thanks for contribution!

Based on what you submitted, your use case is much more complex than changing the attributes of existing concepts (template 5 you used). 

It seems that you want to add links from RxNorm Dose Forms to routes and (?) build a hierarchy.
If that is correct, the process may look as follows:

1) specify the routes you need, making sure that none of them already exist in route domain.
2) add new routes.
3) add links between the routes and dose forms (also think about the name of such link).
4) add hierarchical links among routes.

The complexity of the task exceeds currently supported community contribution, so more discussion will be needed to figure out the best way to proceed (Vocab WG call can be a place for it). It seems that you already built something, so it may be useful to look at your files (and not at templates).

**Comment by tiozab** [comment]
Thank you, @aostropolets , sure, let's discuss this in a community call when @cgreich is back from holidays :-) 
You assumed correctly, we want to add links between dose form and route. 
1. happy to conclude on the new route names together, I think we would want the "old" route names to be deleted to avoid confusion (?) if this is possible..
2. adding new routes (you would be doing this right?)
3. adding the links (you would be doing this right?). We can think of a name together.
4. adding hierarchical links (you would be doing this right?)
I know it is very complex, but very useful :-) We are using this hierarchy for the dose estimation formulas we are working on right now e.g. 
Happy to look at the file together during a call to which @Tsemharb, @cgreich and myself contributed / consented upon. 
Here is it for a pre-view (it is not much more than the template I filled).
[dose form_TB_AG_CR_final.xlsx](https://github.com/OHDSI/Vocabulary-v5.0/files/12181223/dose.form_TB_AG_CR_final.xlsx)
Happy to discuss further during a call.

**Comment by cgreich** [comment]
@he is.

@aostropolets: I think she wants to abolish the existing routes and create a new simplified set of routes plus some hierarchy. And then map RxNorm Dose Forms to them.

Is that correct, @tiozab?

**Comment by tiozab** [comment]
@cgreich I think that is the idea, because the existing routes do not serve any purpose as far as I know

**Comment by aostropolets** [comment]
1. From what I heard people use routes in their CDMs, so such a change definitely requires a ratification (think CDM WG). 
2. What you want to do is more complex than the currently supported use cases (and that's why you had hard time fitting your changes into existing templates). That means we will not be doing it through community contribution in its current form.  Would be happy to discuss how to move forward though. Christian usually doesn't come to Vocab WG calls, so I'd suggest scheduling a separate call.

**Comment by tiozab** [comment]
Ok, I will send out a doodle to have a call, to whom though? @aostropolets @cgreich , anybody else?

**Comment by tiozab** [comment]
@aostropolets @cgreich I made this doodle link, feel free to forward to those who should attend. Please reply by Sep 14th, dates start from Sep 25th.
https://doodle.com/meeting/participate/id/bmypQQpb

**Comment by aostropolets** [comment]
Thanks for the doodle. I don't seem to be able to invite people in the doodle itself. Maybe sending a link through the email would be a better option to ensure that it doesn't get lost here and we can forward it to others. In case you don't have my email it's ostropolets at ohdsi dot org

**Comment by tiozab** [comment]
email sent :-)

**Comment by tiozab** [comment]
@aostropolets @cgreich after the OHDSI US meeting and the decided steps forward, I will use this github issue. I start now with editing the T4 template and will put any questions here.

**Comment by tiozab** [comment]
@aostropolets @cgreich please find the amended T4 template.
[template_4_adding_vocabulary_TB_20231120.xlsx](https://github.com/OHDSI/Vocabulary-v5.0/files/13417648/template_4_adding_vocabulary_TB_20231120.xlsx)
 I overwrote the examples. Please let me know if anything is unclear. 
physician @francescodernie helped with the route matching

**Comment by tiozab** [comment]
answer by Oleg 
I went through your contribution file and noticed some problems:
1) "Complete view example" contains only new concepts to be added and no relationships at all. We introduced this sheet so it would be easier to review concepts and their relationships. Also, it is easier to fill in concept_manual, and concept_relationship_manual from this complete_view table - just copy your content. 
2) concept_relationship_manual_ex is also incomplete. Concept codes (OMOP1) should be in concept_code field, and also other fields (relationship start and end dates, confidence, predicate_id for mappings, mapping_justification) are missing.
3) concept_synonym_example contains NHS Place of service vocabulary
4) vocabulary_example contains NHS Place of service vocabulary
5) Could you please transfer your content from example sheets to _manual sheets?

**Comment by tiozab** [comment]
Thanks @TinyRickC137 
1. I am not sure what needs to be in the "complete view" the column names are duplicated. 
2. I moved the concept codes to the concept code column
3. removed
4. removed
5. all done

**Comment by aostropolets** [comment]
Right, that's my bad with non-informative column names. What complete view serves is to provide is concept_relationship_manual with names in the same view for debugging. for example, now I cannot easily grasp concept_relationship_manual because there are no names in it (but we want to keep its structure as is so that we can easily process it). Here's what we mean (see attached, I added names and other attributes in the complete view and 'New Route' as a placeholder for vocabulary_id). I left SNOMEDs empty - if you could fill the source/target concept_name for SNOMEDs it would be great. And thanks so much for doing it with us!
[template_4_adding_vocabulary_TB_20231120-2.xlsx](https://github.com/OHDSI/Vocabulary-v5.0/files/13428966/template_4_adding_vocabulary_TB_20231120-2.xlsx)

**Comment by tiozab** [comment]
Thanks @aostropolets for the clarifications. I added "target concept name" of dose form and "source concept name" of new routes in the complete view. Hope that is complete now as is :-)
[template_4_adding_vocabulary_TB_20231120-2_TB.xlsx](https://github.com/OHDSI/Vocabulary-v5.0/files/13441300/template_4_adding_vocabulary_TB_20231120-2_TB.xlsx)

**Comment by aostropolets** [comment]
Thanks so much, @tiozab! The next step is to gather community feedback and identify any potential issues with new standard content. To facilitate that, I created a forum post here: https://forums.ohdsi.org/t/new-routes-and-de-standardization-of-old-snomed-routes/20488/3. I saw a couple of questions there already, you would be the best person to answer some them.

**Comment by m-khitrun** [comment]
Hi @tiozab! Thank you for your contribution, it was successfully implemented during this release.
